[receiver/statsd] SIGSEGV when transport other than udp is set

### Component(s)

receiver/statsd

### What happened?

## Description
otelcol-contrib crashes when transport other than `udp` is set for statsd receiver - checked with version 0.77

## Steps to Reproduce
Set `transport` to something different than `udp`
```
receivers:
  statsd:
    transport: "tcp"
exporters:
  logging:
service:
  pipelines:
    metrics:
      exporters:
        - logging
      receivers:
        - statsd
```


## Expected Result
I'd expect otelcol to gracefully shutdown

## Actual Result
Collector crashes with following output:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x108b5ef7c]
```

### Collector version

0.77.0

### Environment information

## Environment
OS: MacOS Ventura version 13.3.1 (a)

### OpenTelemetry Collector configuration

```yaml
receivers:
  statsd:
    transport: "tcp"
exporters:
  logging:
service:
  pipelines:
    metrics:
      exporters:
        - logging
      receivers:
        - statsd
```


### Log output

```shell
2023-05-16T12:43:27.105+0200	info	service/telemetry.go:113	Setting up own telemetry...
2023-05-16T12:43:27.105+0200	info	service/telemetry.go:136	Serving Prometheus metrics	{"address": ":8888", "level": "Basic"}
2023-05-16T12:43:27.105+0200	info	exporter@v0.77.0/exporter.go:286	Development component. May change in the future.	{"kind": "exporter", "data_type": "metrics", "name": "logging"}
2023-05-16T12:43:27.106+0200	info	service/service.go:141	Starting otelcol-contrib...	{"Version": "0.77.0", "NumCPU": 10}
2023-05-16T12:43:27.107+0200	info	extensions/extensions.go:41	Starting extensions...
2023-05-16T12:43:27.107+0200	info	service/service.go:167	Starting shutdown...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x108b5ef7c]

goroutine 1 [running]:
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver.(*statsdReceiver).Shutdown(0x1400085e620, {0x10ae11f80, 0x1})
	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver@v0.77.0/receiver.go:139 +0x2c
go.opentelemetry.io/collector/service/internal/graph.(*Graph).ShutdownAll(0x0?, {0x10b9da7a8, 0x140000740d0})
	go.opentelemetry.io/collector@v0.77.0/service/internal/graph/graph.go:319 +0xd4
go.opentelemetry.io/collector/service.(*Service).Shutdown(0x14000692c80, {0x10b9da7a8, 0x140000740d0})
	go.opentelemetry.io/collector@v0.77.0/service/service.go:173 +0xd0
go.opentelemetry.io/collector/otelcol.(*Collector).setupConfigurationComponents(0x140006bbbc0, {0x10b9da7a8, 0x140000740d0})
	go.opentelemetry.io/collector@v0.77.0/otelcol/collector.go:185 +0x494
go.opentelemetry.io/collector/otelcol.(*Collector).Run(0x140006bbbc0, {0x10b9da7a8, 0x140000740d0})
	go.opentelemetry.io/collector@v0.77.0/otelcol/collector.go:209 +0x3c
go.opentelemetry.io/collector/otelcol.NewCommand.func1(0x140001a7800, {0x108eb4596?, 0x2?, 0x2?})
	go.opentelemetry.io/collector@v0.77.0/otelcol/command.go:38 +0x70
github.com/spf13/cobra.(*Command).execute(0x140001a7800, {0x1400006e1c0, 0x2, 0x2})
	github.com/spf13/cobra@v1.7.0/command.go:940 +0x5c8
github.com/spf13/cobra.(*Command).ExecuteC(0x140001a7800)
	github.com/spf13/cobra@v1.7.0/command.go:1068 +0x35c
github.com/spf13/cobra.(*Command).Execute(0x14000930540?)
	github.com/spf13/cobra@v1.7.0/command.go:992 +0x1c
main.runInteractive({{0x14000930540, 0x140009317a0, 0x14000930960, 0x14000930120, 0x14000931860}, {{0x108edd59b, 0xf}, {0x108f5ee2c, 0x1f}, {0x108eae1e6, ...}}, ...})
	github.com/open-telemetry/opentelemetry-collector-releases/contrib/main.go:31 +0x40
main.run(...)
	github.com/open-telemetry/opentelemetry-collector-releases/contrib/main_others.go:11
main.main()
	github.com/open-telemetry/opentelemetry-collector-releases/contrib/main.go:24 +0x134
```


### Additional context

The example config file does not work as it sets transport to `custom_transport`:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.77.0/receiver/statsdreceiver/testdata/config.yaml

--- a/.chloggen/cj-handle-no-statsd-server.yaml
+++ b/.chloggen/cj-handle-no-statsd-server.yaml
@@ -1,0 +1,20 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: statsdreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Handles StatsD server not running when shutting down to avoid NPE
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [22004]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

--- a/receiver/statsdreceiver/receiver.go
+++ b/receiver/statsdreceiver/receiver.go
@@ -122,7 +122,7 @@ func (r *statsdReceiver) Start(ctx context.Context, host component.Host) error {
 
 // Shutdown stops the StatsD receiver.
 func (r *statsdReceiver) Shutdown(context.Context) error {
-	if r.cancel == nil {
+	if r.cancel == nil || r.server == nil {
 		return nil
 	}
 	err := r.server.Close()

--- a/receiver/statsdreceiver/receiver_test.go
+++ b/receiver/statsdreceiver/receiver_test.go
@@ -83,8 +83,20 @@ func Test_statsdreceiver_Start(t *testing.T) {
 			require.NoError(t, err)
 			err = receiver.Start(context.Background(), componenttest.NewNopHost())
 			assert.Equal(t, tt.wantErr, err)
+
+			assert.NoError(t, receiver.Shutdown(context.Background()))
 		})
 	}
+}
+
+func TestStatsdReceiver_ShutdownBeforeStart(t *testing.T) {
+	ctx := context.Background()
+	cfg := createDefaultConfig().(*Config)
+	nextConsumer := consumertest.NewNop()
+	rcv, err := New(receivertest.NewNopCreateSettings(), *cfg, nextConsumer)
+	assert.NoError(t, err)
+	r := rcv.(*statsdReceiver)
+	assert.NoError(t, r.Shutdown(ctx))
 }
 
 func TestStatsdReceiver_Flush(t *testing.T) {

